sdk/tables/azure-data-tables/tests/test_retry.py

```diff
@@ -97,91 +96,66 @@ def test_retry_on_server_error(self, tables_storage_account_name, tables_primary
     @TablesPreparer()
     def test_retry_on_socket_timeout(self, tables_storage_account_name, tables_primary_storage_account_key):
-        retry = LinearRetry(backoff=1)
         retry_transport = RetryRequestTransport(connection_timeout=11, read_timeout=0.000000000001)
         self._set_up(
             tables_storage_account_name,
             tables_primary_storage_account_key,
-            retry_policy=retry,
             transport=retry_transport,
-            default_table=False)
+            default_table=False,
+            retry_mode=RetryMode.Fixed,
+            retry_backoff_factor=1)
     
         new_table_name = self.get_resource_name('uttable')
         try:
             with pytest.raises(AzureError) as error:
-                self.ts.create_table(new_table_name)
+                self.ts.get_service_properties()
 
             # 3 retries + 1 original == 4
             assert retry_transport.count == 4
             # This call should succeed on the server side, but fail on the client side due to socket timeout
             self.assertTrue('read timeout' in str(error.value), 'Expected socket timeout but got different exception.')
 
         finally:
-            # we must make the timeout normal again to let the delete operation succeed
-            self.ts.delete_table(new_table_name, connection_timeout=(11, 11))
-            self._tear_down(connection_timeout=(11, 11))
-
+            self._tear_down()
 
-    # Waiting on fix to client pipeline
-    # @TablesPreparer()
-    # def test_no_retry(self, tables_storage_account_name, tables_primary_storage_account_key):
-    #     self._set_up(tables_storage_account_name, tables_primary_storage_account_key, retry_total=0, default_table=False)
+    @TablesPreparer()
+    def test_no_retry(self, tables_storage_account_name, tables_primary_storage_account_key):
+        self._set_up(tables_storage_account_name, tables_primary_storage_account_key, retry_total=0, default_table=False)
 
-    #     new_table_name = self.get_resource_name('uttable')
+        new_table_name = self.get_resource_name('uttable')
 
-    #     # Force the create call to 'timeout' with a 408
-    #     callback = ResponseCallback(status=201, new_status=408).override_status
+        # Force the create call to 'timeout' with a 408
+        callback = ResponseCallback(status=201, new_status=500).override_status
 
-    #     try:
-    #         with pytest.raises(HttpResponseError) as error:
-    #             self.ts.create_table(new_table_name, raw_response_hook=callback)
-    #         self.assertEqual(error.value.response.status_code, 408)
-    #         self.assertEqual(error.value.reason, 'Created')
+        try:
+            with pytest.raises(HttpResponseError) as error:
+                self.ts.create_table(new_table_name, raw_response_hook=callback)
+            self.assertEqual(error.value.response.status_code, 500)
+            self.assertEqual(error.value.reason, 'Created')
```

Comment:
```suggestion
             assert error.value.response.status_code == 500
             assert error.value.reason == 'Created'
```
Nit: trying to keep `unittest` out of tests so we can eventually remove it.